File: iosApp/iosApp/Screens/BusinessCalendarScreen.swift
import SwiftUI
import SharedLogic

class CalendarViewModel: ObservableObject {
    @Published var events: [CalendarEvent] = []
    
    init() {
        self.events = [
            CalendarEvent(id: "1", title: "Geçici Vergi Beyannamesi", dateMillis: Int64(Date().timeIntervalSince1970 * 1000 + 86400000 * 2), type: .taxDeadline, isCritical: true, colorHex: "#F44336"),
            CalendarEvent(id: "2", title: "KOSGEB Kadın Girişimci Desteği Kapanış", dateMillis: Int64(Date().timeIntervalSince1970 * 1000 + 86400000 * 5), type: .grantWindow, isCritical: true, colorHex: "#4CAF50"),
            CalendarEvent(id: "3", title: "Mali Müşavirle Evrak Paylaşımı", dateMillis: Int64(Date().timeIntervalSince1970 * 1000), type: .dailyTask, isCritical: false, colorHex: "#FFC107")
        ]
    }
}

struct BusinessCalendarScreen: View {
    @StateObject private var viewModel = CalendarViewModel()
    
    var body: some View {
        NavigationView {
            ScrollView {
                VStack(spacing: 20) {
                    MockCalendarWidgetSwiftUI(events: viewModel.events)
                        .padding()
                        .background(Color(.systemBackground))
                        .cornerRadius(24)
                        .shadow(color: Color.black.opacity(0.1), radius: 5, y: 2)
                        .padding(.horizontal)
                    
                    VStack(alignment: .leading, spacing: 12) {
                        Text("Kritik Görevler")
                            .font(.title3)
                            .fontWeight(.bold)
                            .padding(.horizontal)
                        
                        ForEach(viewModel.events, id: \.id) { event in
                            EventListItemSwiftUI(event: event)
                                .padding(.horizontal)
                        }
                    }
                }
                .padding(.vertical)
            }
            .navigationTitle("İşletme Takvimi")
            .navigationBarTitleDisplayMode(.inline)
            .background(Color(.systemGroupedBackground).edgesIgnoringSafeArea(.all))
        }
    }
}

struct MockCalendarWidgetSwiftUI: View {
    let events: [CalendarEvent]
    
    let daysOfWeek = ["Pzt", "Sal", "Çar", "Per", "Cum", "Cmt", "Paz"]
    
    var body: some View {
        VStack(spacing: 16) {
            Text("Mayıs 2026")
                .font(.headline)
                .fontWeight(.bold)
            
            HStack {
                ForEach(daysOfWeek, id: \.self) { day in
                    Text(day)
                        .font(.caption)
                        .foregroundColor(.gray)
                        .frame(maxWidth: .infinity)
                }
            }
            
            VStack(spacing: 8) {
                // Mock days 1-28
                ForEach(0..<4) { row in
                    HStack {
                        ForEach(1..<8) { col in
                            let dayNumber = row * 7 + col
                            CalendarDayItemSwiftUI(day: "\(dayNumber)", events: events.filter {
                                // Mocking events to show up on 15, 20, 25
                                ($0.id == "1" && dayNumber == 15) || ($0.id == "2" && dayNumber == 20) || ($0.id == "3" && dayNumber == 25)
                            })
                            .frame(maxWidth: .infinity)
                        }
                    }
                }
            }
        }
    }
}

struct CalendarDayItemSwiftUI: View {
    let day: String
    let events: [CalendarEvent]
    
    var body: some View {
        VStack(spacing: 4) {
            Text(day)
                .font(.system(size: 14))
                .fontWeight(day == "15" ? .bold : .regular)
                .frame(width: 30, height: 30)
                .background(day == "15" ? Color.blue.opacity(0.2) : Color.clear)
                .clipShape(Circle())
            
            HStack(spacing: 2) {
                ForEach(events, id: \.id) { event in
                    Circle()
                        .fill(colorForType(event.type))
                        .frame(width: 4, height: 4)
                }
            }
        }
    }
    
    func colorForType(_ type: EventType) -> Color {
        switch type {
        case .taxDeadline: return .red
        case .grantWindow: return .green
        case .dailyTask: return .yellow
        default: return .gray
        }
    }
}

struct EventListItemSwiftUI: View {
    let event: CalendarEvent
    
    var body: some View {
        HStack(spacing: 16) {
            Circle()
                .fill(colorForType(event.type))
                .frame(width: 12, height: 12)
            
            VStack(alignment: .leading, spacing: 4) {
                Text(event.title)
                    .font(.headline)
                    .foregroundColor(event.isCritical ? .primary : .secondary)
                
                Text(formatDate(event.dateMillis))
                    .font(.subheadline)
                    .foregroundColor(.secondary)
            }
            Spacer()
        }
        .padding()
        .background(event.isCritical ? Color.blue.opacity(0.1) : Color(.systemBackground))
        .cornerRadius(16)
    }
    
    func colorForType(_ type: EventType) -> Color {
        switch type {
        case .taxDeadline: return .red
        case .grantWindow: return .green
        case .dailyTask: return .yellow
        default: return .gray
        }
    }
    
    func formatDate(_ millis: Int64) -> String {
        let date = Date(timeIntervalSince1970: TimeInterval(millis) / 1000.0)
        let formatter = DateFormatter()
        formatter.dateFormat = "dd MMM, HH:mm"
        formatter.locale = Locale(identifier: "tr_TR")
        return formatter.string(from: date)
    }
}

File: androidApp/src/main/kotlin/com/anlarsinsoftware/girisimkolay/calendar/ui/BusinessCalendarScreen.kt
package com.anlarsinsoftware.girisimkolay.calendar.ui

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.anlarsinsoftware.girisimkolay.calendar.domain.entity.CalendarEvent
import com.anlarsinsoftware.girisimkolay.calendar.domain.entity.EventType
import com.anlarsinsoftware.girisimkolay.calendar.viewmodel.CalendarViewModel
import org.koin.androidx.compose.koinViewModel
import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun BusinessCalendarScreen(viewModel: CalendarViewModel = koinViewModel()) {
    val events by viewModel.events.collectAsState()

    Scaffold(
        topBar = {
            TopAppBar(
                title = { Text("İşletme Takvimi", fontWeight = FontWeight.Bold) },
                colors = TopAppBarDefaults.topAppBarColors(
                    containerColor = MaterialTheme.colorScheme.background,
                    titleContentColor = MaterialTheme.colorScheme.onBackground
                )
            )
        }
    ) { padding ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(padding)
        ) {
            // Simulated Monthly Calendar Widget
            Card(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(16.dp),
                shape = RoundedCornerShape(24.dp),
                colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surface),
                elevation = CardDefaults.cardElevation(defaultElevation = 4.dp)
            ) {
                MockCalendarWidget(events)
            }

            Text(
                text = "Kritik Görevler",
                fontSize = 20.sp,
                fontWeight = FontWeight.Bold,
                modifier = Modifier.padding(horizontal = 16.dp, vertical = 8.dp)
            )

            LazyColumn(
                contentPadding = PaddingValues(16.dp),
                verticalArrangement = Arrangement.spacedBy(12.dp)
            ) {
                items(events) { event ->
                    EventListItem(event)
                }
            }
        }
    }
}

@Composable
fun MockCalendarWidget(events: List<CalendarEvent>) {
    Column(modifier = Modifier.padding(16.dp)) {
        Text(
            text = "Mayıs 2026",
            fontSize = 18.sp,
            fontWeight = FontWeight.Bold,
            modifier = Modifier.fillMaxWidth(),
            textAlign = TextAlign.Center
        )
        Spacer(modifier = Modifier.height(16.dp))
        
        val daysOfWeek = listOf("Pzt", "Sal", "Çar", "Per", "Cum", "Cmt", "Paz")
        Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceEvenly) {
            daysOfWeek.forEach { day ->
                Text(text = day, fontSize = 12.sp, fontWeight = FontWeight.Medium, color = Color.Gray)
            }
        }
        Spacer(modifier = Modifier.height(8.dp))
        
        // Mocking a few days
        val days = (1..31).toList()
        val weeks = days.chunked(7)
        
        weeks.forEach { week ->
            Row(modifier = Modifier.fillMaxWidth().padding(vertical = 4.dp), horizontalArrangement = Arrangement.SpaceEvenly) {
                week.forEach { day ->
                    val dayEvents = events.filter { 
                        // Mock mapping for demo: event 1 on 15th, event 2 on 20th
                        (it.id == "1" && day == 15) || (it.id == "2" && day == 20) || (it.id == "3" && day == 25)
                    }
                    CalendarDayItem(day.toString(), dayEvents)
                }
                // Pad empty days
                repeat(7 - week.size) {
                    Spacer(modifier = Modifier.size(36.dp))
                }
            }
        }
    }
}

@Composable
fun CalendarDayItem(day: String, events: List<CalendarEvent>) {
    Column(
        horizontalAlignment = Alignment.CenterHorizontally,
        modifier = Modifier.size(36.dp)
    ) {
        Box(
            contentAlignment = Alignment.Center,
            modifier = Modifier
                .size(28.dp)
                .clip(CircleShape)
                .background(if (day == "15") MaterialTheme.colorScheme.primaryContainer else Color.Transparent)
        ) {
            Text(text = day, fontSize = 14.sp, fontWeight = if (day == "15") FontWeight.Bold else FontWeight.Normal)
        }
        
        Row(horizontalArrangement = Arrangement.spacedBy(2.dp), modifier = Modifier.padding(top = 2.dp)) {
            events.forEach { event ->
                val color = when(event.type) {
                    EventType.TAX_DEADLINE -> Color(0xFFE53935) // Red
                    EventType.GRANT_WINDOW -> Color(0xFF4CAF50) // Green
                    EventType.DAILY_TASK -> Color(0xFFFFC107) // Yellow
                }
                Box(modifier = Modifier.size(4.dp).clip(CircleShape).background(color))
            }
        }
    }
}

@Composable
fun EventListItem(event: CalendarEvent) {
    val formatter = SimpleDateFormat("dd MMM, HH:mm", Locale("tr", "TR"))
    val dateString = formatter.format(Date(event.dateMillis))
    
    val indicatorColor = when(event.type) {
        EventType.TAX_DEADLINE -> Color(0xFFE53935)
        EventType.GRANT_WINDOW -> Color(0xFF4CAF50)
        EventType.DAILY_TASK -> Color(0xFFFFC107)
    }

    Card(
        modifier = Modifier.fillMaxWidth(),
        colors = CardDefaults.cardColors(
            containerColor = if (event.isCritical) MaterialTheme.colorScheme.primaryContainer.copy(alpha = 0.3f) else MaterialTheme.colorScheme.surface
        ),
        elevation = CardDefaults.cardElevation(defaultElevation = if (event.isCritical) 4.dp else 1.dp),
        shape = RoundedCornerShape(16.dp)
    ) {
        Row(
            modifier = Modifier.padding(16.dp),
            verticalAlignment = Alignment.CenterVertically
        ) {
            Box(
                modifier = Modifier.size(12.dp).clip(CircleShape).background(indicatorColor)
            )
            Spacer(modifier = Modifier.width(16.dp))
            Column {
                Text(
                    text = event.title,
                    fontWeight = FontWeight.Bold,
                    fontSize = 16.sp,
                    color = MaterialTheme.colorScheme.onSurface
                )
                Text(
                    text = dateString,
                    fontSize = 14.sp,
                    color = MaterialTheme.colorScheme.onSurfaceVariant
                )
            }
        }
    }
}
